test/integration/targets/collections/collection_root_user/ansible_collections/testns/testcoll/plugins/modules/uses_nested_same_as_module.py

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/python
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import sys
+
+from ansible_collections.testns.testcoll.plugins.module_utils.nested_same.nested_same import nested_same
+
+
+def main():
+    mu_result = nested_same.nested_same()
+    print(json.dumps(dict(changed=False, source='user', mu_result=mu_result)))
+
+    sys.exit()
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
I see you're testing both importing the function and importing the module above it using `from [...] import` syntax.  I'm not sure if this code path is walking the ast but if so, it should probably also test:
```
import ansible_collections.testns.testcoll.plugins.module_utils.nested_same.nested_same.nested_same
mu_result = ansible_collections.testns.testcoll.plugins.module_utils.nested_same.nested_same.nested_same.nested_same()
```

Because `import` and `from[...] import` create different ast nodes.